src/Analyzers/CSharp/Tests/UseParameterNullChecking/UseParameterNullCheckingTests.cs

```diff
@@ -0,0 +1,507 @@
+        [InlineData("is")]
+        public async Task TestWithBraces(string @operator)
+        {
+            await TestInRegularAndScriptAsync(
+@"using System;
+
+class C
+{
+    void M(string s)
+    {
+        if ([||]s " + @operator + @" null)
+        {
+            throw new ArgumentNullException(nameof(s));
+        }
+    }
+}",
+@"using System;
+
+class C
+{
+    void M(string s!!)
+    {
+    }
+}", parseOptions: s_parseOptions);
+        }
+
+        [Theory]
+        [Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        [InlineData("==")]
+        [InlineData("is")]
+        public async Task TestLocalFunction(string @operator)
+        {
+            await TestInRegularAndScriptAsync(
+@"using System;
+
+class C
+{
+    void M()
+    {
+        local("""");
+        void local(string s)
+        {
+            if ([||]s " + @operator + @" null)
+            {
+                throw new ArgumentNullException(nameof(s));
+            }
+        }
+    }
+}",
+@"using System;
+
+class C
+{
+    void M()
+    {
+        local("""");
+        void local(string s!!)
+        {
+        }
+    }
+}", parseOptions: s_parseOptions);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestEqualitySwapped()
+        {
+            await TestInRegularAndScriptAsync(
+@"using System;
+
+class C
+{
+    void M(string s)
+    {
+        if ([||]null == (object)s)
+            throw new ArgumentNullException(nameof(s));
+    }
+}",
+@"using System;
+
+class C
+{
+    void M(string s!!)
+    {
+    }
+}", parseOptions: s_parseOptions);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestNotEquality()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"using System;
+
+class C
+{
+    void M(string s)
+    {
+        if ([||](object)s != null)
+            throw new ArgumentNullException(nameof(s));
+    }
+}", parameters: new TestParameters(parseOptions: s_parseOptions));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestNullCoalescingThrow()
+        {
+            await TestInRegularAndScriptAsync(
+@"using System;
+
+class C
+{
+    private readonly string s;
+    void M(string s)
+    {
+        this.s = [||]s ?? throw new ArgumentNullException(nameof(s));
+    }
+}",
+@"using System;
+
+class C
+{
+    private readonly string s;
+    void M(string s!!)
+    {
+        this.s = s;
+    }
+}", parseOptions: s_parseOptions);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestReferenceEqualsCheck()
+        {
+            await TestInRegularAndScriptAsync(
+@"using System;
+
+class C
+{
+    private readonly string s;
+    void M(string s)
+    {
+        if ([||]object.ReferenceEquals(s, null))
```

Comment:
add test with `C.ReferenceEquals(s, null)`  (also are tehre reverse forms of all of these?  e.g. `ReferenceEquals(null, s)`)?

Reply:
I think the current level of testing for the "reverse" form is adequate, but less thorough than the "regular" form with null on the right. I did add a "null on the left" case for ReferenceEquals in response to this feedback, though.